indexing-service/src/main/java/org/apache/druid/indexing/common/task/IndexTask.java

```diff
@@ -850,12 +850,12 @@ private TaskStatus generateAndPublishSegments(
         new FireDepartment(dataSchema, new RealtimeIOConfig(null, null), null);
     FireDepartmentMetrics buildSegmentsFireDepartmentMetrics = fireDepartmentForMetrics.getMetrics();
 
+    final TaskRealtimeMetricsMonitor metricsMonitor = TaskRealtimeMetricsMonitorBuilder.build(
+        this,
+        fireDepartmentForMetrics,
+        buildSegmentsMeters
+    );
     if (toolbox.getMonitorScheduler() != null) {
```

Comment:
Nit: Can `toolbox.getMonitorScheduler` be separated to avoid repetitive usages of this method?